File: cadwiki-nuget/cadwiki.AutoCAD2021.Base.Utilities/Shared/UserInput.vb
Option Strict On
Option Infer Off
Option Explicit On

Imports Autodesk.AutoCAD.ApplicationServices
Imports Autodesk.AutoCAD.DatabaseServices
Imports Autodesk.AutoCAD.EditorInput
Imports Autodesk.AutoCAD.Geometry

Public Class UserInput
    Public Class PointFromUser
        Public Point As Point3d
        Public Success As Boolean
    End Class

    Public Shared Function GetEntityFromUser(doc As Document, promptEntityOptions As PromptEntityOptions) As ObjectId
        Dim entityResult As PromptEntityResult = doc.Editor.GetEntity(promptEntityOptions)
        Dim id As ObjectId = entityResult.ObjectId
        If entityResult.Status = PromptStatus.Cancel Then
            Return Nothing
        End If
        Return id
    End Function

    Public Shared Function GetPointFromUser(doc As Document, promptPointOptions As PromptPointOptions) As PointFromUser
        Dim promptPointResult As PromptPointResult = doc.Editor.GetPoint(promptPointOptions)
        Dim point3d As Point3d = promptPointResult.Value
        Dim pointFromUser As PointFromUser = New PointFromUser
        pointFromUser.Point = point3d
        pointFromUser.Success = True
        If promptPointResult.Status = PromptStatus.Cancel Then
            pointFromUser.Success = False
        End If
        Return pointFromUser
    End Function

    Public Shared Function GetSelectionFromUser(doc As Document, message As String) As SelectionSet
        doc.Editor.WriteMessage(vbLf + message + vbLf)
        Dim promptSelectionResult As PromptSelectionResult = doc.Editor.GetSelection()
        Dim selectionSet As SelectionSet = promptSelectionResult.Value
        If promptSelectionResult.Status = PromptStatus.Cancel Then
            Return Nothing
        End If
        Return selectionSet
    End Function

    Public Shared Function GetSelectionFromUser(doc As Document, filter As SelectionFilter, message As String) _
        As SelectionSet
        doc.Editor.WriteMessage(vbLf + message + vbLf)
        Dim promptSelectionResult As PromptSelectionResult = doc.Editor.GetSelection(filter)
        Dim selectionSet As SelectionSet = promptSelectionResult.Value
        If promptSelectionResult.Status = PromptStatus.Cancel Then
            Return Nothing
        End If
        Return selectionSet
    End Function


    Public Shared Function GetIntegerFromUser(doc As Document, options As PromptDoubleOptions) As Integer
        Dim promptDoubleResult As PromptDoubleResult = doc.Editor.GetDouble(options)
        If promptDoubleResult.Status = PromptStatus.OK Then
            Dim value As Double = promptDoubleResult.Value
            Dim int As Integer = CType(value, Integer)
            Return int
        End If
        Return Nothing
    End Function
End Class

File: cadwiki-nuget/cadwiki.AC/Utilities/UserInput.cs
using System;
using Autodesk.AutoCAD.ApplicationServices;
using Autodesk.AutoCAD.DatabaseServices;
using Autodesk.AutoCAD.EditorInput;
using Autodesk.AutoCAD.Geometry;
using Microsoft.VisualBasic;

namespace cadwiki.AC.Utilities
{

    public class UserInput
    {
        public class PointFromUser
        {
            public Point3d Point;
            public bool Success;
        }

        public static ObjectId GetEntityFromUser(Document doc, PromptEntityOptions promptEntityOptions)
        {
            var entityResult = doc.Editor.GetEntity(promptEntityOptions);
            var id = entityResult.ObjectId;
            if (entityResult.Status == PromptStatus.Cancel)
            {
                return default;
            }
            return id;
        }

        public static PointFromUser GetPointFromUser(Document doc, PromptPointOptions promptPointOptions)
        {
            var promptPointResult = doc.Editor.GetPoint(promptPointOptions);
            var point3d = promptPointResult.Value;
            var pointFromUser = new PointFromUser();
            pointFromUser.Point = point3d;
            pointFromUser.Success = true;
            if (promptPointResult.Status == PromptStatus.Cancel)
            {
                pointFromUser.Success = false;
            }
            return pointFromUser;
        }

        public static SelectionSet GetSelectionFromUser(Document doc, string message)
        {
            doc.Editor.WriteMessage(Environment.NewLine + message + Environment.NewLine);
            var promptSelectionResult = doc.Editor.GetSelection();
            var selectionSet = promptSelectionResult.Value;
            if (promptSelectionResult.Status == PromptStatus.Cancel)
            {
                return null;
            }
            return selectionSet;
        }

        public static SelectionSet GetSelectionFromUser(Document doc, SelectionFilter filter, string message)
        {
            doc.Editor.WriteMessage(Environment.NewLine + message + Environment.NewLine);
            var promptSelectionResult = doc.Editor.GetSelection(filter);
            var selectionSet = promptSelectionResult.Value;
            if (promptSelectionResult.Status == PromptStatus.Cancel)
            {
                return null;
            }
            return selectionSet;
        }


        public static int GetIntegerFromUser(Document doc, PromptDoubleOptions options)
        {
            var promptDoubleResult = doc.Editor.GetDouble(options);
            if (promptDoubleResult.Status == PromptStatus.OK)
            {
                double value = promptDoubleResult.Value;
                int @int = (int)Math.Round(value);
                return @int;
            }
            return default;
        }
    }
}